muikku-core/src/main/java/fi/otavanopisto/muikku/users/WorkspaceUserEntityController.java

```diff
@@ -205,6 +215,38 @@ public WorkspaceUserEntity updateIdentifier(WorkspaceUserEntity workspaceUserEnt
   }
 
   public WorkspaceUserEntity updateActive(WorkspaceUserEntity workspaceUserEntity, Boolean active) {
+    if (Boolean.FALSE.equals(active)) {
+      // #6620: Remove past workspaces from the list of student's last workspaces
+      UserEntity userEntity = userEntityController.findUserEntityByDataSourceAndIdentifier(workspaceUserEntity.getUserSchoolDataIdentifier().getDataSource(), workspaceUserEntity.getUserSchoolDataIdentifier().getIdentifier());
+      UserEntityProperty lastWorkspaces = userEntityController.getUserEntityPropertyByKey(userEntity, "last-workspaces");
+      
+      if (lastWorkspaces != null) {
+        if (StringUtils.isEmpty(lastWorkspaces.getValue())) {
```

Comment:
Should most likely be `if (!StringUtils.isEmpty(lastWorkspaces.getValue()))`

Reply:
Or `StringUtils.isNotEmpty(...`